pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -461,55 +462,64 @@ func (h *upgradeHandler) cleanup(upgrade *harvesterv1.Upgrade, cleanJobs bool) e
 			}
 			jobs, err := h.jobCache.List(plan.Namespace, set.AsSelector())
 			if err != nil {
-				return err
+				return nil, err
 			}
 			for _, job := range jobs {
 				logrus.Debugf("Deleting job %s/%s", job.Namespace, job.Name)
 				if err := h.jobClient.Delete(job.Namespace, job.Name, &metav1.DeleteOptions{}); err != nil {
-					return err
+					return nil, err
 				}
 			}
 		}
 
 		logrus.Debugf("Deleting plan %s/%s", plan.Namespace, plan.Name)
 		if err := h.planClient.Delete(plan.Namespace, plan.Name, &metav1.DeleteOptions{}); err != nil {
-			return err
+			return nil, err
```

Comment:
it deserves to skip the `IsNotFound` error

Reply:
Addressed. Check `deletionTimestamp` and `IsNotFound` error before deleting.